server/src/main/java/mateusz/michal/chat/Registration/Service/RegistrationService.java

```diff
@@ -187,34 +175,42 @@ private boolean isUserInDatabaseByEmail(String email){
         return user.isPresent();
     }
 
-    private boolean isEmailIncorrect(String email){
+    private boolean isEmailCorrect(String email){
         Pattern pattern = Pattern.compile("[a-zA-Z]+[\\.a-zA-Z0-9]*@[a-zA-Z0-9]+\\.[a-z]{2,}[a-z]*");
-        return !pattern.matcher(email).matches();
+        return pattern.matcher(email).matches();
     }
 
-    private boolean isPasswordIncorrect(String password) {
+    private boolean isPasswordStrong(String password) {
         Pattern pattern = Pattern.compile("(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])" +
-                "(?=.*[\\!\\@\\#\\$\\%\\^\\&\\*])(?!=.*\\s).{8,15}");
-        return !pattern.matcher(password).matches();
+                "(?=.*[\\!\\@\\#\\$\\%\\^\\&\\*]).{0,}");
+        return pattern.matcher(password).matches();
+    }
+
+    private boolean isPasswordToShort(String password){
+        return password.length() < 8;
+    }
+
+    private boolean isPassworToLong(String password){
+        return password.length() > 15;
     }
     
     // TODO delete this after test refactoring
     public RegisterFormErrorCode saveUserToDatabase(@NotNull UserDTO userDTO) {
-        if(isNameEmptyString(userDTO.getName())){
+        if(isEmpty(userDTO.getName())){
             return RegisterFormErrorCode.NAME_MISSING;
-        } else if(isEmailEmptyString(userDTO.getEmail())){
+        } else if(isEmpty(userDTO.getEmail())){
             return RegisterFormErrorCode.EMAIL_MISSING;
-        } else if(isPassworEmptyString(userDTO.getPassword())){
+        } else if(isEmpty(userDTO.getPassword())){
             return RegisterFormErrorCode.PASSWORD_MISSING;
         } else if(isNameIncorrect(userDTO.getName())) {
             return RegisterFormErrorCode.NAME_INCORRECT;
         } else if (isUserInDatabaseByName(userDTO.getName())){
             return RegisterFormErrorCode.NAME_OCCUPIED;
         } else if(isUserInDatabaseByEmail(userDTO.getEmail())){
             return RegisterFormErrorCode.EMAIL_OCCUPIED;
-        } else if(isEmailIncorrect(userDTO.getEmail())){
+        } else if(isEmailCorrect(userDTO.getEmail())){
             return RegisterFormErrorCode.EMAIL_INCORRECT;
-        } else if(isPasswordIncorrect(userDTO.getPassword())){
+        } else if(isPasswordStrong(userDTO.getPassword())){
```

Comment:
Tu też - jeśli `isPasswordStrong` zwraca `true` to zwracamy błąd słabego hasła - błąd logiczny :dagger:

Reply:
j.w.